### PR TITLE
Support cirq.I.on_each()

### LINKED ISSUE #1855: cirq.I doesn't have an on_each() method
This is because `IdentityGate` doesn't subclass `SingleQubitGate`.  Instead `I = IdentityGate(num_qubits=1)`.

`cirq.H`, `X`, `T`, etc. all have `for_each()` so I expected `cirq.I` to also have this method.

### PR DESCRIPTION
**Pull request by dabacon**
This only works for single qubit identity.

Fixes #1855

### CODE REVIEW

File: cirq/ops/gate_features.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def on_each(self, *targets: Union[raw_types.Qid, Iterable[Any]]
         Raises:
             ValueError if targets are not instances of Qid or List[Qid].
         """
-        operations = []  # type: List[raw_types.Operation]
+        operations: raw_types.Operation = []
```
Comment (cduck) [inline cirq/ops/gate_features.py]: Why change this in this PR?  Also missing List[...].

File: cirq/ops/common_gates.py [hunk 2]
```diff
@@ -628,6 +628,38 @@ def __init__(self, num_qubits):
     def num_qubits(self) -> int:
         return self._num_qubits
 
+    def on_each(self, *targets: Union[raw_types.Qid, Iterable[Any]]
+               ) -> List[raw_types.Operation]:
+        """Returns a list of operations apply the single qubit identity
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/ops/common_gates.py]: fixed and fixed in SingleQubitGate

File: cirq/ops/common_gates.py [hunk 3]
```diff
@@ -628,6 +628,38 @@ def __init__(self, num_qubits):
...
+        Raises:
+            ValueError if targets are not instances of Qid or List[Qid] or
+            the gate from which this is applied is not a single qubit identity
+            gate.
+        """
+        if self._num_qubits != 1:
+            raise ValueError(
+                'IdentityGate only supports on_each when it is a one qubit '
+                'gate.')
+        operations: raw_types.Operation = []
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/ops/common_gates.py]: fixed

File: cirq/ops/common_gates_test.py [hunk 4]
```diff
@@ -216,6 +216,23 @@ def test_identity_init(num_qubits):
     assert cirq.IdentityGate(num_qubits).num_qubits() == num_qubits
 
 
+def test_identity_on_each():
+    q0, q1, q2 = cirq.LineQubit.range(3)
+    assert cirq.I.on_each(q0, q1, q2) == [cirq.I(q0), cirq.I(q1), cirq.I(q2)]
+    assert cirq.I.on_each([q0, [q1],
+                           q2]) == [cirq.I(q0),
+                                    cirq.I(q1),
+                                    cirq.I(q2)]
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/ops/common_gates_test.py]: added

File: cirq/ops/common_gates_test.py [hunk 5]
```diff
@@ -216,6 +216,23 @@ def test_identity_init(num_qubits):
...
+    with pytest.raises(ValueError):
+        cirq.I.on_each('abc')
+
+
+def test_identity_on_each_only_single_qubit():
+    q0, q1 = cirq.LineQubit.range(2)
+    with pytest.raises(ValueError):
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/ops/common_gates_test.py]: done

File: cirq/ops/gate_features.py (hunk 1 above)
Comment (dabacon) [reply cirq/ops/gate_features.py]: fixed.  I slightly prefer not using the comments

File: cirq/ops/common_gates_test.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (cduck) [reply cirq/ops/common_gates_test.py]: Did you forget to push the change?

**Comment by dabacon** [comment]
oops forgot to push changes, fixing

File: cirq/ops/common_gates_test.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/ops/common_gates_test.py]: fixed

**Comment by mpharrigan** [comment]
Most other gates (MeasurementGate being the exception) operate on a fixed number of qubits. What would be the downside of providing only the single-qubit identity gate.

**Pull request merged**